airbyte-webapp/src/views/Connection/ConnectionForm/components/TransformationField.tsx

```diff
@@ -44,7 +44,10 @@ const TransformationField: React.FC<TransformationFieldProps> = ({
         setEditableItem(idx);
         onStartEdit?.();
       }}
-      onCancel={clearEditableItem}
+      onCancel={() => {
+        clearEditableItem();
+        onEndEdit?.();
```

Comment:
Removing the toggle made me notice a bug: If you clicked outside the Add Transformation modal it wouldn't editing transformation to false. Now it does!